handlers/eth1Token.go

```diff
@@ -72,51 +72,35 @@ func Eth1Token(w http.ResponseWriter, r *http.Request) {
 		logger.WithError(err).Errorf("error generating qr code for address %v", token)
 	}
 
-	if len(metadata.Price) == 0 {
-		metadata.Price = []byte("32.523423")
-	}
-
-	marketCap := float64(0)
 	ethExchangeRate := float64(0)
 	if len(metadata.Price) > 0 && len(metadata.TotalSupply) > 0 {
-		mul := decimal.NewFromFloat(float64(10)).Pow(decimal.NewFromBigInt(new(big.Int).SetBytes(metadata.Decimals), 0))
-		num := decimal.NewFromBigInt(new(big.Int).SetBytes(metadata.TotalSupply), 0)
-
-		priceS := string(metadata.Price)
 		tokenPrice := decimal.New(0, 0)
-		if priceS != "" {
-			var err error
-			tokenPrice, err = decimal.NewFromString(priceS)
-			if err != nil {
-				logger.WithError(err).Errorf("error getting price from string - FormatTokenBalance price: %v", priceS)
-			}
+		if len(metadata.Price) > 0 {
+			tokenPrice = decimal.NewFromBigInt(new(big.Int).SetBytes(metadata.Price), 0)
 		}
 
-		marketCap, _ = tokenPrice.Mul(num.Div(mul)).Float64()
-
-		ethUsdRate := decimal.NewFromFloat(price.GetEthPrice("USD"))
-		logger.Infof("usd rate %s", ethUsdRate)
+		ethUsdRate := decimal.NewFromFloat(price.GetPrice(utils.Config.Frontend.ElCurrencySymbol, "USD"))
 		if !ethUsdRate.IsZero() {
 			ethExchangeRate, _ = tokenPrice.Div(ethUsdRate).Float64()
 		}
 	}
 
 	data := InitPageData(w, r, "blockchain", "/token", fmt.Sprintf("Token 0x%x", token), templateFiles)
+	supply := decimal.NewFromBigInt(new(big.Int).SetBytes(metadata.TotalSupply), 0).Div(decimal.NewFromInt(10).Pow(decimal.NewFromBigInt(new(big.Int).SetBytes(metadata.Decimals), 0)))
```

Comment:
I assume you need a `DivRound(..., metadata.Decimals)` here. (Pseudocode from me)